src/components/dateTimePickerInput.js

```diff
@@ -28,287 +28,265 @@
-    const [errorState, setErrorState] = useState(error);
-    const [afterFirstValidation, setAfterFirstValidation] = useState(false);
-    const helperTextResolved = useText(helperText);
-    const [helper, setHelper] = useState(helperTextResolved);
-    const valueMissingMessage = useText(validationValueMissing);
-    const placeholderText = useText(placeholder);
-    const minValueText = useText(minValue);
-    const maxValueText = useText(maxValue);
-    const beforeMinValueMessage = useText(validationBeforeMinValue);
-    const afterMaxValueMessage = useText(validationAfterMaxValue);
-    const invalidValueMessage = useText(validationInvalidValue);
-    const dataComponentAttributeValue = useText(dataComponentAttribute);
-    const clearable = true;
-    const { current: labelControlRef } = useRef(generateUUID());
 
     const localeMap = {
       nl: nlLocale,
       en: enLocale,
     };
 
-    const parsedLabel = useText(label);
-    const labelText = parsedLabel;
-    const isValidDate = (date) => date instanceof Date && !isNaN(date);
+    const isDev = env === 'dev';
+
+    const valueText = useText(value);
+    const optionHelperText = useText(helperText);
 
-    const convertToDate = (date) => {
-      if (isValidDate(date)) {
-        const dateString = `${date.getFullYear()}-${String(
-          date.getMonth() + 1,
-        ).padStart(2, '0')}-${String(date.getDate()).padStart(2, '0')}`;
+    const validationValueMissingText = useText(validationValueMissing);
+    const validationInvalidValueText = useText(validationInvalidValue);
+    const validationBeforeMinValueText = useText(validationBeforeMinValue);
+    const validationAfterMaxValueText = useText(validationAfterMaxValue);
 
-        return dateString;
-      }
-      return '';
-    };
+    const dataComponentAttributeValue = useText(dataComponentAttribute);
+
+    const { current: labelControlRef } = useRef(generateUUID());
+
+    const [selectedDate, setSelectedDate] = useState(valueText);
+    const [errorState, setErrorState] = useState(error);
+    const [currentHelperText, setHelper] = useState(optionHelperText);
+    const [isDisabled, setIsDisabled] = useState(disabled);
+    const [isFirstValidation, setIsFirstValidation] = useState(true);
+
+    B.defineFunction('Clear', () => setSelectedDate(null));
+    B.defineFunction('Enable', () => setIsDisabled(false));
+    B.defineFunction('Disable', () => setIsDisabled(true));
+
+    const META_API_DATE_FORMAT = 'yyyy-MM-dd';
+    const META_API_TIME_FORMAT = 'HH:mm:ss';
+
+    function getFormat(typeFormat) {
+      return {
+        date: dateFormat || 'dd/MM/yyyy',
+        time: timeFormat || 'HH:mm:ss',
+        datetime: datetimeFormat || 'dd/MM/yyyy HH:mm:ss',
+      }[typeFormat];
+    }
+
+    function isValidDate(date) {
+      return date instanceof Date && !isNaN(date);
+    }
+
+    // useEffect for 'valueText', trigger on first componentRender (and data-loaded)
```

Comment:
remove commented code?

Reply:
it's not commented code, but an explanation about the reponsibility of the function. Since there is no room for a function-name in this setup.